akka-testkit-typed/src/main/scala/akka/testkit/typed/internal/Within.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.testkit.typed.internal
+
+import akka.annotation.InternalApi
+import akka.testkit.typed.TestKitSettings
+
+import scala.concurrent.duration._
+import akka.util.PrettyDuration._
+import akka.testkit.typed.scaladsl._
+
+@InternalApi
+private[akka] trait Within {
+
+  implicit def settings: TestKitSettings
+
+  private var end: Duration = Duration.Undefined
+
+  /**
+   * if last assertion was expectNoMessage, disable timing failure upon within()
```

Comment:
I would like that `expectNoMessage` would not use the `remaining` or `remainingOrDefault` at all.
It should only use the given `max` parameter or the default `ExpectNoMessageDefaultTimeout`.